source/Public/Connect-FabricAccount.ps1

```diff
@@ -49,51 +56,91 @@ function Connect-FabricAccount {
+                    $null = Connect-AzAccount -Tenant $TenantId
+                }
+                else {
+                    # If no TenantId is provided, connect to the default tenant
+                    Write-Message "No TenantId provided, connecting to default tenant..." -Level Verbose
+                    $null = Connect-AzAccount
+                }
+            }
+            $azContext = Get-AzContext
         }
 
-        $azContext = Get-AzContext
-
-        Write-Verbose "Connected: $($azContext.Account)"
+        Write-Message "Connected: $($azContext.Account)" -Level Verbose
 
-        if ($PSCmdlet.ShouldProcess("Setting Fabric authentication token for $($azContext.Account)")) {
+        if ($PSCmdlet.ShouldProcess("Setting Fabric authentication token and headers for $($azContext.Account)")) {
+            Write-Message "Get authentication token from $($FabricSession.ResourceUrl)" -Level Verbose
+            $FabricSession.AccessToken = (Get-AzAccessToken -ResourceUrl $FabricSession.ResourceUrl)
+            $ssPtr = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($FabricSession.AccessToken.Token)
+            $Token = ([System.Runtime.InteropServices.Marshal]::PtrToStringBSTR($ssPtr))
+            Write-Message "Setup headers for Fabric API calls" -Level Debug
+            $FabricSession.HeaderParams = @{'Authorization' = "Bearer {0}" -f $Token }
 
-            Write-Verbose "Get authentication token"
-            $FabricSession.FabricToken = (Get-AzAccessToken -ResourceUrl $FabricSession.ResourceUrl).Token
-            Write-Verbose "Token: $($FabricSession.FabricToken)"
-            }
+            # Copy session values to exposed $FabricConfig
+            $FabricConfig.TenantId = $FabricSession.AccessToken.TenantId
+            $FabricConfig.TokenExpiresOn = $FabricSession.AccessToken.ExpiresOn
+            $FabricConfig.FabricHeaders = $FabricSession.HeaderParams    # Remove this;
+        }
 
-         if ($PSCmdlet.ShouldProcess("Setting Fabric headers for $($azContext.Account)")) {
-            Write-Verbose "Setup headers for API calls"
-            $FabricSession.HeaderParams = @{'Authorization' = "Bearer {0}" -f $FabricSession.FabricToken }
-            Write-Verbose "HeaderParams: $($FabricSession.HeaderParams)"
+        if ($PSCmdlet.ShouldProcess("Setting Azure authentication token and headers for $($azContext.Account)")) {
+            Write-Message "Get authentication token from $($AzureSession.BaseApiUrl)" -Level Verbose
+            $AzureSession.AccessToken = (Get-AzAccessToken -ResourceUrl $AzureSession.BaseApiUrl)
+            $ssPtr = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($AzureSession.AccessToken.Token)
+            $Token = ([System.Runtime.InteropServices.Marshal]::PtrToStringBSTR($ssPtr))
+            Write-Message "Setup headers for Azure API calls" -Level Debug
+            $AzureSession.HeaderParams = @{'Authorization' = "Bearer {0}" -f $Token }
```

Comment:
Wondering whether there is a benefit to doing this rather than just converting from secure string to plain text?

```
$token = $AzureSession.AccessToken.Token | ConvertFrom-SecureString -AsPlainText 
```

calling @SQLDBAWithABeard.... :D